java-manual/modules/ROOT/pages/transactions.adoc

```diff
@@ -0,0 +1,487 @@
+The most commonly needed ones are listed below.
+
+// .An essential list of `Result` methods
+[cols="2,3"]
+|===
+|Method |Description
+
+|`list() List<Record>`
+|Return the remainder of the result as a list.
+
+|`single() Record`
+a|Return the next and only remaining record.
+Calling this method always exhausts the result.
+If more (or less) than one record is available, a `NoSuchRecordException` is raised.
+
+|`next() Record`
+a|Return the next record in the result. Throws `NoSuchRecordException` if no further records are available.
+
+|`hasNext() boolean`
+|Whether the result iterator has a next record to move to.
+
+|`peek() Record`
+|Return the next record from the result _without consuming it_.
+This leaves the record in the buffer for further processing.
+
+|`consume() ResultSummary`
+|Return the query xref:result-summary.adoc[result summary].
+It exhausts the result, so should only be called when data processing is over.
+|===
+
+For a complete list of `Result` methods, see link:https://neo4j.com/docs/api/java-driver/current/org.neo4j.driver/org/neo4j/driver/Result.html[API documentation -- Result].
+
+
+== Session configuration
+
+=== Database selection
+
+It is recommended to *always specify the database explicitly* through the `.withDatabase("<dbName>")` method, even on single-database instances.
+This allows the driver to work more efficiently, as it saves a network round-trip to the server to resolve the home database.
+If no database is given, the link:{neo4j-docs-base-uri}/operations-manual/current/manage-databases/introduction#manage-databases-default[default database] set in the Neo4j instance settings is used.
+
+[source, java]
+----
+// import org.neo4j.driver.SessionConfig;
+
+var session = driver.session(SessionConfig.builder()
+    .withDatabase("neo4j").build());  // mark-line
+----
+
+Do not rely on the link:{neo4j-docs-base-uri}/cypher-manual/current/clauses/use/[`USE`] Cypher clause for database selection with the driver.
+
+
+=== Request routing
+
+In a cluster environment, all sessions are opened in write mode, routing them to the leader.
+You can change this by calling the method `.withRouting(RoutingControl.READ)`.
+Note that `.executeRead()` and `.executeWrite()` automatically override the session's default access mode.
+
+[source, java]
+----
+// import org.neo4j.driver.SessionConfig;
+// import org.neo4j.driver.RoutingControl;
+
+var session = driver.session(SessionConfig.builder()
+    .withDatabase("neo4j")
+    .withRouting(RoutingControl.READ)  // mark-line
+    .build());
+----
+
+[#access-control-note]
+// .Routing ≠ Access control
+[NOTE]
+====
+Although executing a _write_ query in read mode likely results in a runtime error, *you should not rely on this for access control.* The difference between the two modes is that _read_ transactions will be routed to any node of a cluster, whereas _write_ ones will be directed to the leader.
```

Comment:
Same as in the previous file.
Be explicit that this is about routing only, use two sentences and "must not"


Also Neo4j 5 notation is "primary" and "secondary", afaik we don't talk leader/follower anymore.